CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+cmake_minimum_required(VERSION 3.4)
+
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/tools/cmake" ${CMAKE_MODULE_PATH})
+
+project(realm VERSION 2.3.0)
```

Comment:
`git describe` gives something in the form of `<major>-<minor>-<patch>-<commits since tag>-g<sha1(6)>` which is different from the cmake expected `<major>-<minor>-<patch>-<tweak>`. It's also different from what `config.h` expects, even though I would really love to get rid of any reference to a version in the compiled code (I would accept a SHA1)